hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestExternalPathHandling.java

```diff
@@ -0,0 +1,293 @@
+    Assertions.assertEquals(metaClient.getBasePathV2().toString() + "/" + filePath, baseFile.getPath());
+  }
+
+  private void assertEmptyColStats(HoodieBackedTableMetadata hoodieBackedTableMetadata, String partitionPath, String fileName) {
+    Assertions.assertTrue(hoodieBackedTableMetadata.getColumnStats(Collections.singletonList(Pair.of(partitionPath, fileName)), FIELD_1).isEmpty());
+    Assertions.assertTrue(hoodieBackedTableMetadata.getColumnStats(Collections.singletonList(Pair.of(partitionPath, fileName)), FIELD_2).isEmpty());
+  }
+
+  private void assertColStats(HoodieBackedTableMetadata hoodieBackedTableMetadata, String partitionPath, String fileName) {
+    Map<Pair<String, String>, HoodieMetadataColumnStats> field1ColStats = hoodieBackedTableMetadata.getColumnStats(Collections.singletonList(Pair.of(partitionPath, fileName)), FIELD_1);
+    Assertions.assertEquals(1, field1ColStats.size());
+    HoodieMetadataColumnStats column1stats = field1ColStats.get(Pair.of(partitionPath, fileName));
+    Assertions.assertEquals(FIELD_1, column1stats.getColumnName());
+    Assertions.assertEquals(fileName, column1stats.getFileName());
+    Assertions.assertEquals(new IntWrapper(1), column1stats.getMinValue());
+    Assertions.assertEquals(new IntWrapper(2), column1stats.getMaxValue());
+    Assertions.assertEquals(2, column1stats.getValueCount());
+    Assertions.assertEquals(0, column1stats.getNullCount());
+    Assertions.assertEquals(5, column1stats.getTotalSize());
+    Assertions.assertEquals(10, column1stats.getTotalUncompressedSize());
+
+
+    Map<Pair<String, String>, HoodieMetadataColumnStats> field2ColStats = hoodieBackedTableMetadata.getColumnStats(Collections.singletonList(Pair.of(partitionPath, fileName)), FIELD_2);
+    Assertions.assertEquals(1, field2ColStats.size());
+    HoodieMetadataColumnStats column2stats = field2ColStats.get(Pair.of(partitionPath, fileName));
+    Assertions.assertEquals(FIELD_2, column2stats.getColumnName());
+    Assertions.assertEquals(fileName, column2stats.getFileName());
+    Assertions.assertEquals(new StringWrapper("a"), column2stats.getMinValue());
+    Assertions.assertEquals(new StringWrapper("b"), column2stats.getMaxValue());
+    Assertions.assertEquals(3, column2stats.getValueCount());
+    Assertions.assertEquals(1, column2stats.getNullCount());
+    Assertions.assertEquals(10, column2stats.getTotalSize());
+    Assertions.assertEquals(20, column2stats.getTotalUncompressedSize());
+  }
+
+  private JavaRDD<WriteStatus> createRdd(List<WriteStatus> writeStatuses) {
+    return jsc.parallelize(writeStatuses, 1);
+  }
+
+  private WriteStatus createWriteStatus(String commitTime, String partitionPath, String filePath, String fileId) {
+    WriteStatus writeStatus = new WriteStatus();
+    writeStatus.setFileId(fileId);
+    writeStatus.setPartitionPath(partitionPath);
+    HoodieDeltaWriteStat writeStat = new HoodieDeltaWriteStat();
```

Comment:
DeltaWriteStat is used only in Append use cases where log files are added. I am sure you are going this route so that we can test col stats as well. 
I am ok w/ this test set up. but just calling it out.

Reply:
Is there any reason why the col stats are limited to the append only case? What's the rationale for this limitation?

Reply:
The reason I'm testing this out is because we don't want to recompute col stats when we already know their values

Reply:
got it. makes sense. for base files, we do on-demand lookup from parquet metadata to populate values for col stats